OasisDEX/oasis-borrow
# Add internal mixpanel api endpoint

# [Mixpanel Proxy](https://app.shortcut.com/oazo-apps/story/2591/mixpanel-add-tracking-of-transactions-address-changes-that-circumvents-track-blockers-like-brave)
  
## Changes 👷‍♀️
- introduce internal endpoint `/api/mixpanel` to omit blockers like built in Brave
- install `mixpanel` dependency, (note: I've tried using `mixpanel-browser` on endpoint but it didn't work, I guess underlying implementation for sending differs)
- refactor a little bit needed events in `analytics`
- refactor a little bit adding development mode for mixpanel both in browser and server
  
## How to test 🧪
- test if browsers like Brave are not blocking our internal API
![image](https://user-images.githubusercontent.com/19193499/135259246-1e547327-979b-46c4-add3-43c80b8fb32d.png)
- test `accountChange` and some `ConfirmTransaction` events in our UI and see if they are both logged in `development` mode or send to mixpanel in `staging` or `production` mode
![image](https://user-images.githubusercontent.com/19193499/135259477-ad800a9a-8b1d-405e-bbbd-ea5ed4f69665.png)
    
## Definition of done ✔️
